Sub/Src/ComputerVision/libs/pose_calculation.py

```diff
@@ -0,0 +1,126 @@
+    and give us accurate judgements on how far our objects are from the front of the sub.
+    '''
+    def __init__(self, detection, x, y, w, h):
+        '''
+        Initializes my distance calculator
+        Params:
+            detection: Detection object. Includes image with yolo's bouding box drawn around
+            x: x coordinate of the bounding box
+            y: y coordinate of the bounding box
+            w: width of the bounding box
+            h: height of the bounding box
+        Returns:
+            N/A
+        '''
+        self.detection = detection
+        self.x_coordinate = float(x)
+        self.y_coordinate = float(y)
+        self.width = float(w)
+        self.height = float(h)
+        self.camera_matrix = None
+        self.distortion_matrix = None
+        self.two_dim_points = None
+        self.three_dim_points = None
+
+        configs = MechOS_Network_Configs(MECHOS_CONFIG_FILE_PATH)._get_network_parameters()
+
+        self.param_serv = mechos.Parameter_Server_Client(configs["param_ip"], configs["param_port"])
+        self.param_serv.use_parameter_database(configs["param_server_path"])
+
+    def set_matrices(self):
+        '''
+        This function sets our camera, distortion, and dimension points depending on the detection
+        Params:
+            N/A
+        Returns:
+            N/A
+        '''
+
+        self.focal_length_x = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/focal_length_x"))
+        self.focal_length_y = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/focal_length_y"))
+        self.optical_center_x = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/optical_center_x"))
+        self.optical_center_y = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/optical_center_y"))
+
+        self.camera_matrix = np.array([[self.focal_length_x, 0.0, self.optical_center_x],
+                                      [0.0, self.focal_length_y, self.optical_center_y],
+                                      [0.0, 0.0, 1.0]])
+        #print(self.camera_matrix)
+
+        self.distort_var_1 = float(self.param_serv.get_param("Vision/solvePnP/distortion_matrix/k1"))
+        self.distort_var_2 = float(self.param_serv.get_param("Vision/solvePnP/distortion_matrix/k2"))
+        self.distort_var_3 = float(self.param_serv.get_param("Vision/solvePnP/distortion_matrix/p1"))
+        self.distort_var_4 = float(self.param_serv.get_param("Vision/solvePnP/distortion_matrix/p2"))
+        self.distort_var_5 = float(self.param_serv.get_param("Vision/solvePnP/distortion_matrix/k3"))
+
+        self.distortion_matrix = np.array([[self.distort_var_1, self.distort_var_2, self.distort_var_3, self.distort_var_4, self.distort_var_5]])
+        #print(self.distortion_matrix)
+
+        label = self.detection[0]
+
+        if(label == b'Dice 1' or label == b'Dice 2' or label == b'Dice 3' or label == b'Dice 4' or label == b'Dice 5' or label == b'Dice 6'):
```

Comment:
I mentioned this in my slack comment, but you could retrain the network with every dice just being a "Dice" instead of individual dice. I got better results doing this for detecting the dice, however you obviously can't differentiate them. But for testing that helped me out a lot.